waspls/src/Wasp/LSP/Core.hs

```diff
@@ -0,0 +1,46 @@
+module Wasp.LSP.Core
+  ( ServerM,
+    Severity (..),
+    State,
+    ServerConfig,
+  )
+where
+
+import Control.Monad.Trans.Except (ExceptT)
+import Control.Monad.Trans.State.Strict (StateT)
+import Data.Aeson
+import Data.Aeson.Types (prependFailure, typeMismatch)
+import Data.Default (Default (def))
+import Data.Text (Text)
+import Language.LSP.Server (LspT)
+
+type ServerM =
+  ExceptT (Severity, Text) (StateT State (LspT ServerConfig IO))
+
+-- | Error severity levels
```

Comment:
What are these errors actually? They are errors returned by LS? What is their relationship with "diagnose" stuff? Those are not the same, right? I guess when we return compiler errors as "diagnosis", that is a success for LS and LS is not considered to have thrown any errors, right? So these are LS-related errors, not Wasp source related errors?

Reply:
Correct, these are language server errors, separate from diagnostics that get reported to the client.

Reply:
Aha great! I commented above how we should somehow document this, so I will close this thread.